atc/db/team.go

```diff
@@ -163,6 +164,25 @@ func (t *team) FindWorkerForVolume(handle string) (Worker, bool, error) {
 	}))
 }
 
+func (t *team) FindWorkersForResourceCache(rcId int) ([]Worker, error) {
+	return getWorkers(
+		t.conn, workersQuery.
+			Join("volumes v ON v.worker_name = w.name").
+			Join("worker_resource_caches wrc ON v.worker_resource_cache_id = wrc.id").
+			Join("resource_caches rc ON rc.id = wrc.resource_cache_id").
+			Where(sq.And{
+				sq.Eq{"rc.id": rcId},
+				sq.Eq{"w.state": "running"},
```

Comment:
nit: It might be nice to use the `WorkerStateRunning` const within this query https://github.com/concourse/concourse/blob/0e8560d9b2ca285870a752a43fa0954642ac7297/atc/db/worker.go#L32